### PR TITLE
Added Extractor Documentation

### PR DESCRIPTION
**Pull request by OfficialKris**
## Description

Added extractor docs information as it took me several hours to figure out the first time :pensive: 

## Testing

Ran with `npm run docs:dev`

### CODE REVIEW

**Review by Snowiiii — COMMENTED** [review]
You don't need to copy the Particle.kt file when adding a new Extractor

**Comment by OfficialKris** [comment]
That was the way I did it. If someone else knows how to do it from scratch, they could add some information about it.

File: docs/developer/extractor.md [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+4. Wait until the last extractor completes to cancel the server
+
+5. The generated files will be under `./run/pumpkin_extractor_output`
+
+### Add an Extractor
+
+1. The simplest way to add a new extractor is to copy one (for example the `Particles.kt` file)
+2. Rename the file to your new registry
+3. Modify the file to add the required information according the internal structure of the Minecraft `Registry`
```
Comment (Snowiiii) [inline docs/developer/extractor.md:40]: You just need a File which implements the Extractor interface. Place it in the `extractors` folder and register it in the Main class. Done

**Comment by Snowiiii** [comment]
I also have to say that the Extractor is very easy and i don't really see why you spend several hours finding out how it works

**Comment by OfficialKris** [comment]
Yea it was easy to add a new one but hard to figure out how to run it in the first place. I never used gradle or kotlin before so I was very confused and tried many things.

**Pull request closed without merging**